### PR TITLE
Switched xml file config to OrientDb

### PR DESCRIPTION
**Pull request by ivanursul**

### CODE REVIEW

File: strongbox-storage/strongbox-storage-api/log_2016-05-05.0.log [hunk 1]
```diff
@@ -0,0 +1,230 @@
...
+23:46:35.898 05-05-2016 | INFO  | main                 | o.c.strongbox.configuration.ConfigurationManager   |     -> Repository: group-releases-nested
+23:46:35.898 05-05-2016 | INFO  | main                 | o.c.strongbox.configuration.ConfigurationManager   |     -> Repository: group-releases-nested-deep-1
+23:46:35.898 05-05-2016 | INFO  | main                 | o.c.strongbox.configuration.ConfigurationManager   |     -> Repository: group-releases-nested-deep-2
+23:46:35.899 05-05-2016 | INFO  | main                 | o.c.strongbox.configuration.ConfigurationManager   |     -> Repository: group-releases-nested-deep-3
+23:46:35.907 05-05-2016 | INFO  | main                 | o.c.s.services.impl.ArtifactResolutionServiceImpl  | Loading resolvers...
+23:46:35.907 05-05-2016 | INFO  | main                 | o.c.s.services.impl.ArtifactResolutionServiceImpl  |  -> class org.carlspring.strongbox.storage.resolvers.FSLocationResolver
+23:46:35.907 05-05-2016 | INFO  | main                 | o.c.s.services.impl.ArtifactResolutionServiceImpl  |  -> class org.carlspring.strongbox.storage.resolvers.GroupLocationResolver
```
Comment (carlspring) [inline strongbox-storage/strongbox-storage-api/log_2016-05-05.0.log]: @ivanursul : This log file seems to have been added by mistake?

Comment (ivanursul) [inline strongbox-storage/strongbox-storage-api/log_2016-05-05.0.log]: ready

File: strongbox-storage/strongbox-storage-api/src/main/java/org/carlspring/strongbox/configuration/ConfigurationManager.java [hunk 2]
```diff
@@ -136,13 +131,12 @@ public String getRepositoryId(String storageAndRepositoryId)
...
+        return configurationRepository.getConfiguration();
     }
 
-    public Resource getConfigurationResource()
-            throws IOException
-    {
-        return configurationResourceResolver.getConfigurationResource("repository.config.xml", "etc/conf/strongbox.xml");
+    @Override
+    public Resource getConfigurationResource() throws IOException {
+        return null;
```
Comment (carlspring) [inline strongbox-storage/strongbox-storage-api/src/main/java/org/carlspring/strongbox/configuration/ConfigurationManager.java]: @ivanursul,

I think this should be as it was...
The idea is to be able to import a `strongbox.xml` if one is on the path, or specified as via the parameter. If there is no such specified (and there is no configuration yet in OrientDB), then load the one from the classpath (available in the `strongbox-storage-api-resources` artifact).

File: strongbox-storage/strongbox-storage-api/src/main/java/org/carlspring/strongbox/configuration/ConfigurationRepository.java [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+            Configuration content = getConfiguration();
+
+            if (content == null) {
+                createSettings(propertyKey);
+            }
+        }
+    }
+
+    private void createSettings(String propertyKey) {
+        if (System.getProperty(propertyKey) != null)
```
Comment (carlspring) [inline strongbox-storage/strongbox-storage-api/src/main/java/org/carlspring/strongbox/configuration/ConfigurationRepository.java]: I'm not sure I'm following what this method does, would you mind elaborating...?

File: strongbox-storage/strongbox-storage-indexing/src/main/java/org/carlspring/strongbox/services/impl/ConfigurationManagementServiceImpl.java [hunk 4]
```diff
@@ -87,18 +91,20 @@ public void setProxyConfiguration(String storageId,
                                       ProxyConfiguration proxyConfiguration)
             throws IOException, JAXBException
     {
+        Configuration configuration = configurationManager.getConfiguration();
         if (storageId != null && repositoryId != null)
         {
-            configurationManager.getConfiguration()
+            configuration
```
Comment (carlspring) [inline strongbox-storage/strongbox-storage-indexing/src/main/java/org/carlspring/strongbox/services/impl/ConfigurationManagementServiceImpl.java]: (A cosmetic fix): Would you mind re-indenting this the way it was before?

File: strongbox-storage/strongbox-storage-indexing/src/main/java/org/carlspring/strongbox/services/impl/ConfigurationManagementServiceImpl.java [hunk 5]
```diff
@@ -189,12 +201,17 @@ public void removeRepositoryFromAssociatedGroups(String repositoryId)
...
         {
+            Configuration configuration = configurationManager.getConfiguration();
+
             for (Repository repository : includedInGroupRepositories)
             {
-                configurationManager.getConfiguration()
-                                    .getStorage(repository.getStorage().getId())
-                                    .getRepository(repository.getId())
-                                    .getGroupRepositories().remove(repositoryId);
+                configuration
```
Comment (carlspring) [inline strongbox-storage/strongbox-storage-indexing/src/main/java/org/carlspring/strongbox/services/impl/ConfigurationManagementServiceImpl.java]: (A cosmetic fix): Would you mind re-indenting this the way it was before?

File: strongbox-storage/strongbox-storage-indexing/src/main/java/org/carlspring/strongbox/services/impl/StorageManagementServiceImpl.java [hunk 6]
```diff
@@ -51,6 +51,7 @@ public void removeStorage(String storageId)
         for (Repository repository : storage.getRepositories().values())
         {
             repositoryManagementService.removeRepository(storageId, repository.getId());
+
```
Comment (carlspring) [inline strongbox-storage/strongbox-storage-indexing/src/main/java/org/carlspring/strongbox/services/impl/StorageManagementServiceImpl.java]: Seems like an unnecessary line... perhaps a typo?

File: strongbox-testing/strongbox-testing-core/src/main/java/org/carlspring/strongbox/testing/TestCaseWithArtifactGeneration.java [hunk 7]
```diff
@@ -21,6 +21,7 @@
 /**
  * @author mtodorov
  */
+
```
Comment (carlspring) [inline strongbox-testing/strongbox-testing-core/src/main/java/org/carlspring/strongbox/testing/TestCaseWithArtifactGeneration.java]: Seems like an unnecessary line... perhaps a typo?

File: strongbox-web-core/src/main/java/org/carlspring/strongbox/rest/app/spring/security/ConfigurationRepository.java [hunk 8]
```diff
@@ -0,0 +1,5 @@
+package org.carlspring.strongbox.rest.app.spring.security;
+
+public class ConfigurationRepository {
```
Comment (carlspring) [inline strongbox-web-core/src/main/java/org/carlspring/strongbox/rest/app/spring/security/ConfigurationRepository.java]: I'm not quite sure I get how you're using this empty class... Is Spring doing some weaving/instrumentation, or what?

File: strongbox-storage/strongbox-storage-api/src/main/java/org/carlspring/strongbox/configuration/ConfigurationRepository.java (hunk 3 above)
Comment (ivanursul) [inline strongbox-storage/strongbox-storage-api/src/main/java/org/carlspring/strongbox/configuration/ConfigurationRepository.java]: It's saving settings from strongbox.xml

File: strongbox-storage/strongbox-storage-indexing/src/main/java/org/carlspring/strongbox/services/impl/ConfigurationManagementServiceImpl.java (hunk 4 above)
Comment (ivanursul) [inline strongbox-storage/strongbox-storage-indexing/src/main/java/org/carlspring/strongbox/services/impl/ConfigurationManagementServiceImpl.java]: It's not a cosmetical fix - I'm getting settings from db, and updating them

File: strongbox-storage/strongbox-storage-indexing/src/main/java/org/carlspring/strongbox/services/impl/ConfigurationManagementServiceImpl.java (hunk 5 above)
Comment (ivanursul) [inline strongbox-storage/strongbox-storage-indexing/src/main/java/org/carlspring/strongbox/services/impl/ConfigurationManagementServiceImpl.java]: It's not a cosmetical fix - I'm getting settings from db, and updating them

File: strongbox-web-core/src/main/java/org/carlspring/strongbox/rest/app/spring/security/ConfigurationRepository.java (hunk 8 above)
Comment (ivanursul) [inline strongbox-web-core/src/main/java/org/carlspring/strongbox/rest/app/spring/security/ConfigurationRepository.java]: I'm not sure If I was the author of this class. Removed.

File: strongbox-storage/strongbox-storage-indexing/src/main/java/org/carlspring/strongbox/services/impl/ConfigurationManagementServiceImpl.java (hunk 4 above)
Comment (carlspring) [inline strongbox-storage/strongbox-storage-indexing/src/main/java/org/carlspring/strongbox/services/impl/ConfigurationManagementServiceImpl.java]: I meant, change the line from:
    
    configuration
                                 .getStorage(storageId)

To:

    configuration.getStorage(storageId)

and so on...

(The same applies to the similar comments below).

File: strongbox-commons/src/main/java/org/carlspring/strongbox/db/DbUtils.java [hunk 9]
```diff
@@ -0,0 +1,66 @@
...
+            }
+            else
+            {
+                db.create();
+            }
+            return code.apply(db);
+        }
+        finally
+        {
+/*            if (!db.isClosed())
```
Comment (carlspring) [inline strongbox-commons/src/main/java/org/carlspring/strongbox/db/DbUtils.java]: I see both `finally` blocks (this one and the one a bit further down below) are commented out. Is the code there needed, or not...?

File: strongbox-storage/strongbox-storage-api/src/test/java/org/carlspring/strongbox/storage/resolvers/GroupLocationResolverTest.java [hunk 10]
```diff
@@ -117,8 +112,16 @@ public void testGroupIncludesWithOutOfServiceRepository()
     {
         System.out.println("# Testing group includes with out of service repository...");
 
-        configurationManager.getConfiguration().getStorage("storage0")
-                            .getRepository("releases").putOutOfService();
+        /*Repository repository = configurationManager.getConfiguration().getStorage("storage0")
```
Comment (carlspring) [inline strongbox-storage/strongbox-storage-api/src/test/java/org/carlspring/strongbox/storage/resolvers/GroupLocationResolverTest.java]: Could this be removed?

**Comment by AlexOreshkevich** [comment]
Hi Martin, this branch was merged to the latest strongbox master branch code, please review it and incorporate.

**Comment by carlspring** [comment]
@AlexOreshkevich,

This was merged into @korest 's fork where he did some work before it was merged into the master in [48c02b64](https://github.com/strongbox/strongbox/commit/48c02b641e9c3aca3dfc45babeefd00da399f774) via pull #138 .

**Comment by carlspring** [comment]
@ivanursul ,

Thanks for your work on this!

**Pull request closed without merging**